test/integration/src/test/java/io/pravega/test/integration/BackUpRecoveryTest.java

```diff
@@ -0,0 +1,857 @@
+        log.info("Created a long term storage.");
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (this.dataLogFactory != null) {
+            this.dataLogFactory.close();
+        }
+        timer.set(0);
+        FileHelpers.deleteFileOrDirectory(this.baseDir);
+    }
+
+    @Override
+    protected int getThreadPoolSize() {
+        return 100;
+    }
+
+    /**
+     * Creates a client to read and write events.
+     */
+    private static class ClientRunner implements AutoCloseable {
+        private final ConnectionFactory connectionFactory;
+        private final ClientFactoryImpl clientFactory;
+        private final ReaderGroupManager readerGroupManager;
+
+        ClientRunner(LocalServiceStarter.ControllerRunner controllerRunner) {
+            this.connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder()
+                    .controllerURI(controllerRunner.getControllerURI()).build());
+            this.clientFactory = new ClientFactoryImpl(SCOPE, controllerRunner.getController(), connectionFactory);
+            this.readerGroupManager = new ReaderGroupManagerImpl(SCOPE, controllerRunner.getController(), clientFactory);
+        }
+
+        @Override
+        public void close() {
+            this.readerGroupManager.close();
+            this.clientFactory.close();
+            this.connectionFactory.close();
+        }
+    }
+
+    /**
+     * Tests the data recovery scenario with just one segment container.
+     *  What test does, step by step:
+     *  1.  Starts Pravega locally with just one segment container.
+     *  2.  Writes {@link #TOTAL_NUM_EVENTS} events.
+     *  3.  Shuts down the controller
+     *  4.  Waits for DurableLog to be entirely flushed to the long term storage.
+     *  5.  Shuts down segment store and bookeeper/zookeeper.
+     *  6.  Starts new instance of bk, zk, store and controller.
+     *  7. Reads all events to verify that auto recovery is done and reads are successful.
+     * @throws Exception    In case of an exception occurred while execution.
+     */
+    @Test(timeout = 90000)
+    public void testDurableDataLogRecoverySingleContainer() throws Exception {
+        testRecovery(1, 1, false);
+    }
+
+    /**
+     * Tests the data recovery scenario with multiple segment containers.
+     *  What test does, step by step:
+     *  1.  Starts Pravega locally with just one segment container.
+     *  2.  Writes {@link #TOTAL_NUM_EVENTS} events.
+     *  3.  Shuts down the controller
+     *  4.  Waits for DurableLog to be entirely flushed to the long term storage.
+     *  5.  Shuts down segment store and bookeeper/zookeeper.
+     *  6.  Starts new instance of bk, zk, store and controller.
+     *  7. Reads all events.
+     * @throws Exception    In case of an exception occurred while execution.
+     */
+    @Test(timeout = 180000)
```

Comment:
Here and elsewhere in this file:
This is going to run on every single checkin. Right now we run 1000+ tests in ~20 minutes. If these are taking anywhere near three minutes each, they need to be speed up.